src/main/java/seedu/address/model/pet/AttendanceHashMap.java

```diff
@@ -75,6 +75,15 @@ public void addAttendance(AttendanceEntry attendanceEntry) {
         return Optional.ofNullable(attendanceHashMap.get(attendanceDate));
     }
 
+    /**
+     * Retrieves the attendance entry at the specified date.
+     * @param attendanceDate the specified date.
+     * @return an AttendanceEntry object containing the attendance corresponding entry.
+     */
+    public AttendanceEntry getAttendanceEntry(LocalDate attendanceDate) {
```

Comment:
It is possible to use the `getAttendance` method in `AttendanceHashMap` that returns the Optional!

Reply:
I initially tried with `getAttendance` and run into errors. When I studied the Attendance classes closely I realised that since I am retrieving data from the storage I will never have a null `AttendanceEntry` as it will be an instance of `MissingAttendanceEntry`. Therefore I added this additionally method as it simply the comparator classes.

Reply:
`MissingAttendanceEntry` is only used when needed by the GUI under `AttendanceUtil#getPastWeekAttendance` and is never used in `AttendanceHashMap`, so I don't think it was the cause of your error. Possible to relook?

Reply:
Agreed. Got a bit confused by the `MissingAttendanceEntry`, fixed it. Thanks for the clarification.

Reply:
@Dineshraj555 By the way, remember to remove the additional `getAttendance` method you added if it is no longer needed! Thanks!

Reply:
Done!